File: 4_Matricies/rotare/90MAntiClockwise.js

// space complexity n +  ; time complexity n 
function MRotate90AntiClockwise(array) {
    let arrayB = Array.from(Array(array.length), () => new Array(array.length));
    console.log(array)
    let i;
    for (let y = array.length-1, i = 0; y >= 0;i++, y--) {
      for (let x = array.length-1; x >= 0; x--) {
        arrayB[y][x] = array[x][i];
      }
    }
  
    for (let y = array.length-1; y >= 0; y--) {
      for (let x = array.length-1; x >= 0; x--) {
        array[y][x] = arrayB[y][x];
      }
    }
  }

File: 4_Matricies/rotare/90MAntiClockwise.c
void MRotate90AntiClockwise(int array[4][4], const int SIZE) {
    int arrayB[SIZE][SIZE];
    int i;
    for (int y = SIZE-1, i = 0; y >= 0;i++, y--) 
        for (int x = SIZE-1; x >= 0; x--) 
            arrayB[y][x] = array[x][i];
    for (int y = SIZE-1; y >= 0; y--) 
        for (int x = SIZE-1; x >= 0; x--) 
            array[y][x] = arrayB[y][x];
}